### DISCUSSION TITLE
From<&'a [A]> for CowArray?

### DISCUSSION THREAD

**Discussion by hombit** [rust-ndarray/ndarray#1036] [Q&A] [answered]
Is there any reason why CowArray doesn't implement all `From` which is implemented by both `Array` and `ArrayView`? My current usage is limited to build CowArray from both `&Vec` and `Vec`, but generally I'd like to write something like:

```
struct S<'a> {
    ca: CowArray<'a, f64, Ix1>
}

impl<'a> S<'a> {
    fn new(a: impl Into<CowArray<'a, f64, Ix1>>) -> Self {
        Self {ca: a.into()}
    }
}
```

**Comment by jturner314** [comment, marked as answer]
I've converted this into issue #1037.

**Reply by hombit** [reply]
Thank you! I hasn't been sure that it is not some language limitation or a library design decision
